ray-curator/ray_curator/stages/modules/filter.py

```diff
@@ -0,0 +1,500 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from collections.abc import Callable
+from dataclasses import dataclass
+
+import fasttext
+import huggingface_hub
+import pandas as pd
+from transformers import AutoTokenizer
+
+from ray_curator.backends.base import NodeInfo, WorkerMetadata
+from ray_curator.stages.base import ProcessingStage
+from ray_curator.stages.filters.doc_filter import DocumentFilter
+from ray_curator.tasks import DocumentBatch
+
+
+@dataclass
+class Score(ProcessingStage[DocumentBatch, DocumentBatch]):
```

Comment:
Split into a different file? Since file is called filter.py and the stage is Score?